ts/Core/Axis/Axis.ts

```diff
@@ -7927,8 +7927,20 @@ class Axis {
         erase(chart.axes, this);
         erase((chart as any)[key], this);
 
-        if (isArray((chart.options as any)[key])) {
-            (chart.options as any)[key].splice(this.options.index, 1);
+        const options: Highcharts.AxisOptions[] = (chart.options as any)[key];
+
+        if (isArray(options) && defined(this.options.index)) {
```

Comment:
Please explain in a comment what is going on here, and add a reference to the issue. Both for the sake of the PR, and for the sake of ourselves finding this piece of code in the future. It might be that we should fix this issue in other parts of the code, but under any circumstance this is so hard to grasp that we need comments to explain it to us. As for alternative fixes, check out what we're doing with the `isInternal` option in other parts of the code base.